File: clients/node/sdk/models/tokenResponse.ts
import {AdditionalDataHolder, Parsable, ParseNode, SerializationWriter} from '@microsoft/kiota-abstractions';

export class TokenResponse implements AdditionalDataHolder, Parsable {
    /** The access_token property */
    private _access_token?: string | undefined;
    /** Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well. */
    private _additionalData: Record<string, unknown>;
    /** The created_at property */
    private _created_at?: number | undefined;
    /** The scope property */
    private _scope?: string | undefined;
    /** The token_type property */
    private _token_type?: string | undefined;
    /**
     * Gets the access_token property value. The access_token property
     * @returns a string
     */
    public get access_token() {
        return this._access_token;
    };
    /**
     * Sets the access_token property value. The access_token property
     * @param value Value to set for the access_token property.
     */
    public set access_token(value: string | undefined) {
        this._access_token = value;
    };
    /**
     * Gets the additionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
     * @returns a Record<string, unknown>
     */
    public get additionalData() {
        return this._additionalData;
    };
    /**
     * Sets the additionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
     * @param value Value to set for the AdditionalData property.
     */
    public set additionalData(value: Record<string, unknown>) {
        this._additionalData = value;
    };
    /**
     * Instantiates a new TokenResponse and sets the default values.
     */
    public constructor() {
        this._additionalData = {};
    };
    /**
     * Gets the created_at property value. The created_at property
     * @returns a integer
     */
    public get created_at() {
        return this._created_at;
    };
    /**
     * Sets the created_at property value. The created_at property
     * @param value Value to set for the created_at property.
     */
    public set created_at(value: number | undefined) {
        this._created_at = value;
    };
    /**
     * The deserialization information for the current model
     * @returns a Record<string, (node: ParseNode) => void>
     */
    public getFieldDeserializers() : Record<string, (node: ParseNode) => void> {
        return {
            "access_token": n => { this.access_token = n.getStringValue(); },
            "created_at": n => { this.created_at = n.getNumberValue(); },
            "scope": n => { this.scope = n.getStringValue(); },
            "token_type": n => { this.token_type = n.getStringValue(); },
        };
    };
    /**
     * Gets the scope property value. The scope property
     * @returns a string
     */
    public get scope() {
        return this._scope;
    };
    /**
     * Sets the scope property value. The scope property
     * @param value Value to set for the scope property.
     */
    public set scope(value: string | undefined) {
        this._scope = value;
    };
    /**
     * Serializes information the current object
     * @param writer Serialization writer to use to serialize this model
     */
    public serialize(writer: SerializationWriter) : void {
        if(!writer) throw new Error("writer cannot be undefined");
        writer.writeStringValue("access_token", this.access_token);
        writer.writeNumberValue("created_at", this.created_at);
        writer.writeStringValue("scope", this.scope);
        writer.writeStringValue("token_type", this.token_type);
        writer.writeAdditionalData(this.additionalData);
    };
    /**
     * Gets the token_type property value. The token_type property
     * @returns a string
     */
    public get token_type() {
        return this._token_type;
    };
    /**
     * Sets the token_type property value. The token_type property
     * @param value Value to set for the token_type property.
     */
    public set token_type(value: string | undefined) {
        this._token_type = value;
    };
}

File: clients/python/sdk/models/token_response.py
from __future__ import annotations
from kiota_abstractions.serialization import AdditionalDataHolder, Parsable, ParseNode, SerializationWriter
from typing import Any, Callable, Dict, List, Optional, Union

class TokenResponse(AdditionalDataHolder, Parsable):
    @property
    def access_token(self,) -> Optional[str]:
        """
        Gets the access_token property value. The access_token property
        Returns: Optional[str]
        """
        return self._access_token
    
    @access_token.setter
    def access_token(self,value: Optional[str] = None) -> None:
        """
        Sets the access_token property value. The access_token property
        Args:
            value: Value to set for the access_token property.
        """
        self._access_token = value
    
    @property
    def additional_data(self,) -> Dict[str, Any]:
        """
        Gets the additionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
        Returns: Dict[str, Any]
        """
        return self._additional_data
    
    @additional_data.setter
    def additional_data(self,value: Dict[str, Any]) -> None:
        """
        Sets the additionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
        Args:
            value: Value to set for the AdditionalData property.
        """
        self._additional_data = value
    
    def __init__(self,) -> None:
        """
        Instantiates a new TokenResponse and sets the default values.
        """
        # Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
        self._additional_data: Dict[str, Any] = {}

        # The access_token property
        self._access_token: Optional[str] = None
        # The created_at property
        self._created_at: Optional[int] = None
        # The scope property
        self._scope: Optional[str] = None
        # The token_type property
        self._token_type: Optional[str] = None
    
    @property
    def created_at(self,) -> Optional[int]:
        """
        Gets the created_at property value. The created_at property
        Returns: Optional[int]
        """
        return self._created_at
    
    @created_at.setter
    def created_at(self,value: Optional[int] = None) -> None:
        """
        Sets the created_at property value. The created_at property
        Args:
            value: Value to set for the created_at property.
        """
        self._created_at = value
    
    @staticmethod
    def create_from_discriminator_value(parse_node: Optional[ParseNode] = None) -> TokenResponse:
        """
        Creates a new instance of the appropriate class based on discriminator value
        Args:
            parseNode: The parse node to use to read the discriminator value and create the object
        Returns: TokenResponse
        """
        if parse_node is None:
            raise Exception("parse_node cannot be undefined")
        return TokenResponse()
    
    def get_field_deserializers(self,) -> Dict[str, Callable[[ParseNode], None]]:
        """
        The deserialization information for the current model
        Returns: Dict[str, Callable[[ParseNode], None]]
        """
        fields = {
            "access_token": lambda n : setattr(self, 'access_token', n.get_str_value()),
            "created_at": lambda n : setattr(self, 'created_at', n.get_int_value()),
            "scope": lambda n : setattr(self, 'scope', n.get_str_value()),
            "token_type": lambda n : setattr(self, 'token_type', n.get_str_value()),
        }
        return fields
    
    @property
    def scope(self,) -> Optional[str]:
        """
        Gets the scope property value. The scope property
        Returns: Optional[str]
        """
        return self._scope
    
    @scope.setter
    def scope(self,value: Optional[str] = None) -> None:
        """
        Sets the scope property value. The scope property
        Args:
            value: Value to set for the scope property.
        """
        self._scope = value
    
    def serialize(self,writer: SerializationWriter) -> None:
        """
        Serializes information the current object
        Args:
            writer: Serialization writer to use to serialize this model
        """
        if writer is None:
            raise Exception("writer cannot be undefined")
        writer.write_str_value("access_token", self.access_token)
        writer.write_int_value("created_at", self.created_at)
        writer.write_str_value("scope", self.scope)
        writer.write_str_value("token_type", self.token_type)
        writer.write_additional_data_value(self.additional_data)
    
    @property
    def token_type(self,) -> Optional[str]:
        """
        Gets the token_type property value. The token_type property
        Returns: Optional[str]
        """
        return self._token_type
    
    @token_type.setter
    def token_type(self,value: Optional[str] = None) -> None:
        """
        Sets the token_type property value. The token_type property
        Args:
            value: Value to set for the token_type property.
        """
        self._token_type = value
    

